### ISSUE TITLE
[Bug]: NodePos.setAttribute doesn't re-render NodeView

### ISSUE THREAD

**Issue by rChaoz** [ueberdosis/tiptap#5671]
### Affected Packages

core

### Version(s)

2.7.3

### Bug Description

Using `NodePos.setAttribute()` to update the attributes of a node doesn't seem to re-render a `NodeView` (or the attributes aren't changed at all). Changing it for a custom transaction works.

This doesn't work:
```ts
editor.$node("image", { loadingId }).setAttribute(someAttributes)
```

This works:
```ts
editor.commands.command(({ tr, state }) => {
    let done = false
    state.doc.descendants((node, pos) => {
        if (node.type.name === "image" && node.attrs.loadingId === loadingId) {
            tr.setNodeMarkup(pos, undefined, someAttributes)
            done = true
        }
        return !done
    })
    return true
}),
```

Might be caused by #5672. I don't see any errors in the console tho, like index out of bounds (if I manually call `tr.setNodeMarkup` with `NodePos.pos` as the position it throws that error, as the image is the last node).

### Browser Used

Firefox

### Expected Behavior

-

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by Eshchock1** [comment]
+1

**Comment by nperez0111** [comment]
This is released with v2.9.0
